backend/gateway/config.go

```diff
@@ -183,6 +184,27 @@ type validator interface {
 	Validate() error
 }
 
+// Returns maximum timeout, where 0 is considered maximum (i.e. no timeout).
+func computeMaximumTimeout(cfg *gatewayv1.Timeouts) time.Duration {
+	if cfg == nil {
+		return timeouts.DefaultTimeout
+	}
+
+	ret := cfg.Default.AsDuration()
+	for _, e := range cfg.Overrides {
```

Comment:
How many timeout overrides can we have? In this case if we have more than one type of timeout the maximum is always selected? 

If this is only used for max global timeout why not just have an explicit config option for that then?

Reply:
we can have as many as someone wants to put in. yes, this is just a backstop so the maximum is always selected.

if we added global max (btw this only applies to http.Server so it might be pretty confusing) then we would need to verify that no override exceeds global max, and the user would basically just be performing this algorithm manually and then setting the global max to that value.

Reply:
Cool that makes sense, thanks for the explanation.